src/libs/actions/Report.js

```diff
@@ -507,16 +558,12 @@ function updateReportWithNewAction(reportID, reportAction) {
 
     // If chat report receives an action with IOU, update IOU object
     if (reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU) {
-        const chatReport = lodashGet(allReports, reportID);
-        const iouReportID = lodashGet(chatReport, 'iouReportID');
-        if (iouReportID) {
-            fetchIOUReport(iouReportID, reportID)
-                .then(iouReportObject => setLocalIOUReportData(iouReportObject, reportID));
-        } else if (!chatReport || chatReport.participants.length === 1) {
-            fetchIOUReportID(chatReport ? chatReport.participants[0] : reportAction.actorEmail)
-                .then(iouID => fetchIOUReport(iouID, reportID))
-                .then(iouReportObject => setLocalIOUReportData(iouReportObject, reportID));
-        }
+        const iouReportID = reportAction.originalMessage.IOUReportID;
+
+        // This iouReport is open, so after fetching the report we must update the chatReport link. We can be sure that
+        // the iouReport is open, because reportActions of type CONST.REPORT.ACTIONS.TYPE.IOU can only be triggered for
+        // open iouReports -- see function for more info.
```

Comment:
NAB, rather than do "see function for more info" we should either skip this comment or change it to something like this...

>We know this iouReport is open because reportActions of type CONST.REPORT.ACTIONS.TYPE.IOU can only be triggered for open iouReports. An "open" iouReport means that [explain]. After fetching the report we must update the chatReport link because [explain why].

Reply:
// We know this iouReport is open because reportActions of type CONST.REPORT.ACTIONS.TYPE.IOU can only be
// triggered for open iouReports (an open iouReport has an IOU, but is not yet paid). After fetching the
// iouReport we must update the chatReport link, ensuring that it points to the correct iouReportID. If this
// link update didn't occur, then new IOUs would not be displayed and paid IOUs would show as unpaid.

Reply:
>We know this iouReport is open because reportActions of type CONST.REPORT.ACTIONS.TYPE.IOU can only be triggered for an open iouReport (an open iouReport has an IOU, but is not yet paid). After fetching the iouReport we must update the chatReport with the correct iouReportID. If we don't, then new IOUs would not be displayed and paid IOUs would show as unpaid.

Reply:
I think I forgot to resolve this one, the comment has been updated.

Reply:
I lost this thread last night, but have added your additional changes 👍